lib/ansible/modules/cloud/amazon/code_build.py

```diff
@@ -0,0 +1,338 @@
+    if not isinstance(name, str):
+        module.fail_json(msg="Params was missing name", exception=traceback.format_exc())
+    clean_params = dict((k, v) for k, v in params.items() if v is not None)
+    clean_params.pop('region', None)
+    clean_params.pop('state', None)
+    clean_params.pop('validate_certs', None)
+    formatted_params = snake_dict_to_camel_dict(clean_params)
+
+    # Check if project with that name aleady exists and if so update existing:
+    found = describe_project(client=client, name=name, module=module)
+
+    if 'name' in found:
+        found_last_modified = found['lastModified']
+        resp = update_project(client=client, params=formatted_params, module=module)
+        update_last_modified = resp['project']['lastModified']
+
+        if update_last_modified > found_last_modified:
+            resp['updated'] = True
+        return resp
+    # Or create new project:
+    try:
+        resp = client.create_project(**formatted_params)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable create project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to create project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def update_project(client, params, module):
+    name = params['name']
+
+    try:
+        resp = client.update_project(**params)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable update project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to update project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def delete_project(client, name, module):
+    try:
+        resp = client.delete_project(name=name)
+        return resp
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Unable delete project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Unable to delete project {0}: {1}".format(name, to_native(e)),
+                         exception=traceback.format_exc())
+
+
+def describe_project(client, name, module):
+    project = {}
+    try:
+        projects = client.batch_get_projects(names=[name])['projects']
+        if len(projects) > 0 and isinstance(projects[0], dict):
+            project = projects[0]
+        return project
+    except botocore.exceptions.ClientError as e:
+        return project
```

Comment:
Agreed, call fail_json here. If there's a specific exception that should be masked you can handle it by checking the error code.